include/circt/Dialect/LTL/LTLOps.td

```diff
@@ -0,0 +1,312 @@
+    third, `[a, (b && c), d]` in short.
+
+    By making the delay between concatenated sequences explicit, the `concat`
+    operation behaves nicely in the presence of zero-length sequences. An empty,
+    zero-length sequence in a concatenation behaves as if the sequence wasn't
+    present at all. Compare this to SVAs which struggle with empty sequences.
+    For example, `x ##1 y ##1 z` would become `x ##2 z` if `y` was empty.
+    Similarly, expressing zero or more repetitions of a sequence, `w ##[*]`, is
+    challenging in SVA since concatenation always implies a cycle of delay, but
+    trivial if the delay is made explicit. This is related to the handling of
+    empty rules in a parser's grammar.
+
+    Note that concatenating two boolean values *a* and *b* is equivalent to
+    computing the logical AND of them. Booleans are sequences that check if the
+    boolean is true in the current cycle, which means that the sequence starts
+    and ends in the same cycle. Since concatenation aligns the sequences such
+    that end time of *a* and start time of *b* coincide, the resulting sequence
+    checks if *a* and *b* both are true in the current cycle, which is an AND
+    operation.
+  }];
+}
+
+//===----------------------------------------------------------------------===//
+// Properties
+//===----------------------------------------------------------------------===//
+
+def NotOp : LTLOp<"not", [Pure]> {
+  let arguments = (ins LTLAnyPropertyType:$input);
+  let results = (outs LTLPropertyType:$result);
+  let assemblyFormat = [{
+    $input attr-dict `:` type($input)
+  }];
+
+  let summary = "A negation of a property.";
+  let description = [{
+    Negates the `$input` property. The resulting property evaluates to true if
+    `$input` evaluates to false, and it evaluates to false if `$input` evaluates
+    to true.
+  }];
+}
+
+def ImplicationOp : LTLOp<"implication", [Pure]> {
+  let arguments = (ins LTLAnySequenceType:$antecedent,
+                       LTLAnyPropertyType:$consequent);
+  let results = (outs LTLPropertyType:$result);
+  let assemblyFormat = [{
+    operands attr-dict `:` type(operands)
+  }];
+
+  let summary = "Only check a property after a sequence matched.";
+  let description = [{
+    Preconditions the checking of the `$consequent` property on the
+    `$antecedent` sequence. In a nutshell, if the `$antecedent` sequence matches
+    at a given point in time, the `$consequent` property is checked starting at
+    the point in time at which the matched sequence ends. The result property of
+    the `ltl.implication` holds if the `$consequent` holds. Conversely, if the
+    `$antecedent` does *not* match at a given point in time, the result property
+    trivially holds. This is conceptually identical to the implication operator
+    →, but with additional temporal semantics.
+  }];
+}
+
+def EventuallyOp : LTLOp<"eventually", [Pure]> {
```

Comment:
Hmmm I'm not sure about that. I guess you're right if you are treating sequences and properties as one thing. But if you separate the building blocks for composing sequences from propositions you can verify, a "sequence with potentially unbounded delay" is different from the quantifier "eventually this will hold". Especially on the SV side, the eventually here is a *strong* eventually, meaning that the property has to hold after a finite number of cycles, whereas the unbounded delay is *weak* in the sense that it holds as true when the delay is infinite and the delayed property never materializes.

Reply:
It might be worth noting in the description this is a strong eventually.